### PR TITLE
Sw dspdc 838 transform libraries

### PR DESCRIPTION
**Pull request by snwessel**
Adding the transformation for libraries.

Also removing a few fields from the library table:
- `source`, which is rarely used in encode 
- `product_id` and `lot_id` are both related to the purchase of biosamples, and would make more sense to store in the biosample table. I will create a follow-up ticket to add them there.

### CODE REVIEW

File: transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/LibraryTransformations.scala [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      id = CommonTransformations.readId(libraryInput),
+      crossReferences = libraryInput.read[Array[String]]("dbxrefs"),
+      timeCreated = libraryInput.read[OffsetDateTime]("date_created"),
+      award = libraryInput.read[String]("award"),
+      lab = libraryInput.read[String]("lab"),
+      queriedRnpSizeRange = libraryInput.tryRead[String]("queried_RNP_size_range"),
+      rnaIntegrityNumber = libraryInput.tryRead[Double]("rna_integrity_number"),
+      sizeRange = libraryInput.tryRead[String]("size_range"),
+      strandSpecificity = libraryInput.read[Boolean]("strand_specificity"),
+      treatments = libraryInput.read[Array[String]]("treatments"),
```
Comment (danxmoran) [inline transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/LibraryTransformations.scala]: Do we trust that this array will always be populated in the source data?

File: transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/LibraryTransformations.scala [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      submittedBy = libraryInput.read[String]("submitted_by"),
+      spikeIns = libraryInput.read[Array[String]]("spikeins_used"),
+      biosampleId = libraryInput.read[String]("biosample")
```
Comment (danxmoran) [inline transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/LibraryTransformations.scala]: This value needs transformed to strip the `/biosamples/` prefix for the join against the `biosample` table to work.

File: transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/LibraryTransformations.scala (hunk 1 above)
Comment (snwessel) [reply transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/LibraryTransformations.scala]: From what I've seen, the extracted json always has `"treatments": []` when there are no treatments. I haven't seen any cases where the field is not defined, if that's what you're asking. We could change this to a `tryRead`, but would need to update it elsewhere too (we are using `read` to extract arrays of strings in the other transformations).

Comment (danxmoran) [reply transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/LibraryTransformations.scala]: I'm ok leaving it until we see a problem in practice

**Comment by danxmoran** [comment]
Heads-up: merging master will pull in new scalafmt settings, you might hit a failure for malformatted code if you do it in the browser

**Pull request merged**